assemble-and-restart

```diff
@@ -4,12 +4,14 @@ set -eo pipefail
 
 # We now run the assembly script. If there is a custom one written in the
 # source files, we use that instead.
-if [ -f /opt/app-root/src/.s2i/bin/assemble ]; then
-    /opt/app-root/src/.s2i/bin/assemble
-elif [ -f /usr/local/s2i/assemble ]; then # java s2i
+if [ -f ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble ]; then
+    ${ODO_S2I_SRC_BIN_PATH}/.s2i/bin/assemble
+elif [ -n "${ODO_S2I_SCRIPTS_URL}" ]; then # For S2I scripts path, use the env var set by odo if not available in component source
     rm -rf /opt/app-root/src/.git # ensure we don't copy git files since they can cause problems
-    cp -a /opt/app-root/src/. /tmp/src # java s2i expects the source to be under /tmp/src
-    /usr/local/s2i/assemble
+    if [ "${ODO_S2I_SRC_BIN_PATH}" != "/opt/app-root/src" ]; then
+      cp -a /opt/app-root/src/ ${ODO_S2I_SRC_BIN_PATH} # Copy over the sources from code volume to s2i expected src path
```

Comment:
We talk about this with Anmol but we didn't emphasize it since it would make the odo changes even larger.
Furthermore, we would also have to update the volume, wouldn't we?

Reply:
yes we would, I just commented on https://github.com/redhat-developer/odo/pull/924#pullrequestreview-173155754

Reply:
I think we even added this issue earlier in day to the cabal issues to discuss 😁